Tasks/Docker/containerbuild.ts

```diff
@@ -44,8 +45,8 @@ export function run(connection: ContainerConnection): any {
         command.arg(["--build-arg", buildArgument]);
     });
 
-    var imageName = tl.getInput("imageName", true);
-    imageName = imageUtils.changeDefaultImageNameToLowerCase(imageName);
+    let imageNames = utils.getImageNames(/* multipleImageNamesAllowed */ false);
```

Comment:
The problem with attempting to do this is how to describe this in the task definition.  The imageName property has a visibleRule that includes this restriction: "imageNameInputMode = SingleImageName".  The problem is if we don't want to allow image file input for the Build action, we'd need to hide the imageNameInputMode property when the Build action is selected.  But if you do that, it breaks the visibility of the imageName property.  Because imageNameInputMode is not visible it apparently doesn't have its default value of SingleImageName.  But it needs to have that value in order for the imageName property to be visible based on its rule.  So the result of this is that the user selects the Build action and the image name property never appears.  Because of this limitation, I went ahead with implementing support for file-based input, but limiting it to a single image name.  I don't see this as a big issue.  It adds value for more dynamic scenarios.

---
In reply to: [136702366](https://github.com/Microsoft/vsts-tasks/pull/5076#discussion_r136702366) [](ancestors = 136702366)

Reply:
I think an input which is not visible should retain its default value, but I am not fully sure. Let me also check with @RoopeshNair regarding the inputs changes as he had specific thoughts in mind when we last talked.

Reply:
I was wrong, it retains default value only when parent input is data source binded.

Roopesh had a suggestion to instead add a new action called "Push" multiple images.